api/v1alpha1/oadp_types.go

```diff
@@ -406,8 +406,11 @@ func (dpa *DataProtectionApplication) AutoCorrect() {
 		if pvOperationTimeout, err := time.ParseDuration(resticTimeout); err == nil && dpa.Spec.Configuration.Velero.Args.PodVolumeOperationTimeout == nil {
 			dpa.Spec.Configuration.Velero.Args.PodVolumeOperationTimeout = &pvOperationTimeout
 		}
-	}
-	if dpa.Spec.Configuration.Velero.Args != nil && dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities == "" {
+		if dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities == "" {
+			dpa.Spec.Configuration.Velero.Args.RestoreResourcePriorities = "securitycontextconstraints,customresourcedefinitions,namespaces,storageclasses,volumesnapshotclass.snapshot.storage.k8s.io,volumesnapshotcontents.snapshot.storage.k8s.io,volumesnapshots.snapshot.storage.k8s.io,datauploads.velero.io,persistentvolumes,persistentvolumeclaims,serviceaccounts,secrets,configmaps,limitranges,pods,replicasets.apps,clusterclasses.cluster.x-k8s.io,endpoints,services,-,clusterbootstraps.run.tanzu.vmware.com,clusters.cluster.x-k8s.io,clusterresourcesets.addons.cluster.x-k8s.io"
+		}
+	} else {
+		dpa.Spec.Configuration.Velero.Args = &server.Args{}
```

Comment:
I think you want this line above line 399, because now it's not nil, which means the things done after line 399 are needed here. In fact, if you do that, you don't need the second set of RestoreResourcePriorities (which gives us 2 places to make changes when we update the list).
Basically above line 399, you want:
```
if dpa.Spec.Configuration.Velero.Args == nil {
	dpa.Spec.Configuration.Velero.Args = &server.Args{}
}
```
I think if you do this, the rest of this PR is unnecessary. Also, once that's done, you no longer need the nil check in line 399, so that code can move one indentation level to the left.

Reply:
Done, @shawn-hurley suggested the same thing :)

Reply:
@mateusoliveira43 Yes, this looks just right now.